### PR TITLE
Adding cpputest, docopt and mbed-cloud-client as submodules

### PR DESCRIPTION
**Pull request by prakharbhatt007**
# Mbed Edge pull request

`
Added sub-modules: cpputest, docopt.c
Added .gitignore file.
`

## Description

`1. Added submodules needed to build the edge-core unit build: cpputest, docopt.c
 2. Added .git ignore file
 3. Change in CMake file to create the unit test build
 4. Added files needed by docopt.c to build the  test build of edge-core` 
## Test instructions

`Submitter: Please add instructions to test your PR here.`

`If your pull request is a fix for a defect please write instructions
to reproduce the defect.`

### API change(s)

 - [ ] Not applicable.
 - [ ] API is backwards compatible.
 - [ ] API documentation is updated.

### Example applications updated

 - [ ] Not applicable.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -10,3 +10,12 @@
 [submodule "lib/mbedtls"]
 	path = lib/mbedtls
 	url = https://github.com/ARMmbed/mbedtls
+[submodule "lib/mbed-cloud-client"]
+	path = lib/mbed-cloud-client
+	url = https://github.com/PelionIoT/mbed-cloud-client
```
Comment (ygoyal18) [inline .gitmodules:15]: @prakharbhatt007 mbed-cloud-client is not a submodule as of now. Should we also convert that to submodule?

File: .gitmodules [hunk 2]
```diff
@@ -10,3 +10,12 @@
...
+[submodule "lib/cpputest"]
+	path = lib/cpputest
+	url = git@github.com:cpputest/cpputest.git
```
Comment (OPpuolitaival) [inline .gitmodules:18]: Please use https format
```suggestion
	url = https://github.com/cpputest/cpputest.git
```

File: .gitmodules [hunk 3]
```diff
@@ -10,3 +10,12 @@
...
+[submodule "lib/docopt.c"]
+	path = lib/docopt.c
+	url = git@github.com:docopt/docopt.c.git
```
Comment (OPpuolitaival) [inline .gitmodules:21]: please use https link for public repositories
```suggestion
	url = https://github.com/docopt/docopt.c.git
```

File: .gitmodules (hunk 1 above)
Comment (prakharbhatt007) [reply .gitmodules:15]: yes.. I think we should convert that to submodule.

**Review by ygoyal18 — APPROVED** [review]
LGTM

**Pull request merged**